### ISSUE TITLE
Can't get -1/-2 switches to work

### ISSUE THREAD

**Issue by gkuenning** [pdfpc/pdfpc#528]
**Expected Behaviour**

The -1 and -2 switches are supposed to allow selection of the proper monitor in multi-monitor situations.  For example:

bow:2:779> pdfpc -M
Monitors: 3
 0: * DP-4      [1920x1080+0+0@60Hz     scale=100%]
 1:   DP-2      [1920x1080+3840+0@60Hz  scale=100%]
 2:   HDMI-0    [1920x1080+1920+0@60Hz  scale=100%]
bow:2:780> pdfpc -1 HDMI-0 -2 DP-2 class03_filesystems_beamer.pdf
bow:2:781> pdfpc -2 HDMI-0 -1 DP-2 class03_filesystems_beamer.pdf

Command 780 ought to show the presenter window on the middle screen (monitor 2, HDMI-0) and the presentation on the right-hand screen (monitor 1, DP-1).  Command 780 should reverse those, showing the presenter on the right-hand screen and the presentation in the middle (monitor 2, HDMI-0).

**Actual Behaviour**

No matter what the switches contain, the presenter screen shows up on the screen the cursor is in, and the presentation appears on monitor 2, HDMI-0.

**Steps to reproduce**

On a 3-monitor system, invoke pdfpc with -1 asking for a monitor different from where the cursor is, and the presentation to be on the cursor monitor.

pdfpc version: 4.3.0, current github release
Used Distribution (GTK version, vala version): Vala 0.38.8, gtk 3.22.30
Display Server (X/Wayland): X
HiDPI Setting (yes/no): no

I tried stepping through the code with gdb and couldn't spot the problem.  The switches are recognized, the proper monitor indices are recorded, and the constructors are being called with what seems to be the proper monitors.  I'm not 100% sure but I seem to remember that it even didn't execute the code that picks the monitor that the cursor is in.  

Debugging suggestions would be appreciated...  (Can I use printfs?  I'm not familiar with vala.)

**Comment by fnevgeny** [comment]
If only two monitors are connected, is everything OK?

> Can I use printfs?

E.g., `GLib.printerr()`

**Comment by gkuenning** [comment]
This problem turns out to be an interaction with my window manager (sawfish).  The behavior I described above was partly caused by a matching rule I created long ago, which explicitly forces the presentation window to the second screen.  (I think I did that because at the time the -1 and -2 switches weren't available.)

However, when I remove that rule from my window-manager configuration, both windows wind up on the same screen, which is always the screen that contains my cursor.  I suspect that's because pdfpc creates the windows and then moves them to their final locations, and my window manager is deciding to ignore the move request.  I'll have to dig deeper to find out why.  In the meantime I'm going to leave this issue open just so there's a record of what's going on, but at this point I'm pretty sure it's not a pdfpc bug.
